marker_api/src/ast/stmt.rs

```diff
@@ -79,23 +132,67 @@ impl<'ast> LetStmt<'ast> {
     }
 }
 
+impl_stmt_data!(LetStmt<'ast>, Let);
+
 #[cfg(feature = "driver-api")]
 impl<'ast> LetStmt<'ast> {
     pub fn new(
-        id: LetStmtId,
-        span: SpanId,
+        data: CommonStmtData<'ast>,
         pat: PatKind<'ast>,
         ty: Option<SynTyKind<'ast>>,
         init: Option<ExprKind<'ast>>,
         els: Option<ExprKind<'ast>>,
     ) -> Self {
         Self {
-            id,
-            span,
+            data,
             pat,
             ty: ty.into(),
             init: init.into(),
             els: els.into(),
         }
     }
 }
+
+#[repr(C)]
+#[derive(Debug)]
+pub struct ExprStmt<'ast> {
+    data: CommonStmtData<'ast>,
+    expr: ExprKind<'ast>,
+}
+
+impl<'ast> ExprStmt<'ast> {
+    pub fn expr(&self) -> ExprKind<'ast> {
+        self.expr
+    }
+}
+
+impl_stmt_data!(ExprStmt<'ast>, Expr);
+
+#[cfg(feature = "driver-api")]
+impl<'ast> ExprStmt<'ast> {
+    pub fn new(data: CommonStmtData<'ast>, expr: ExprKind<'ast>) -> Self {
+        Self { data, expr }
+    }
+}
+
+#[repr(C)]
+#[derive(Debug)]
+pub struct ItemStmt<'ast> {
+    data: CommonStmtData<'ast>,
+    item: ItemKind<'ast>,
+}
+
+impl<'ast> ItemStmt<'ast> {
+    pub fn item(&self) -> ItemKind<'ast> {
+        self.item
+    }
+}
+
+impl_stmt_data!(ItemStmt<'ast>, Item);
+
+#[cfg(feature = "driver-api")]
+impl<'ast> ItemStmt<'ast> {
+    pub fn new(data: CommonStmtData<'ast>, item: ItemKind<'ast>) -> Self {
+        Self { data, item }
+    }
+}
```

Comment:
Could it make sense to just mark the fields of the structs conditionally `pub` instead?

Reply:
The `visibility` crate sadly doesn't work on field structs. Otherwise, that could be an option. I considered adding something like [`impl_new`](https://crates.io/crates/impl_new) to cfg derive these things. I was a bit hesitant with Marker's FFI types, but I just saw that it uses `impl Into<Ty>` that could actually work :thinking:

Reply:
I wouldn't add a proc macro dependency to `marker_api`. I love the fact that it can be a zero dependency crate, meaning the lint crates can also be effectively zero-dep (except for `marker_api`), and compile blazingly fast

Reply:
The macro would only be needed when it's compiled with the `driver-api` feature, so it would remain zero-dep :D

Reply:
Ah, right!

Reply:
I'm undecided on this one. [impl_new](https://crates.io/crates/impl_new) looks great but only has a total of 102 downloads. [derive-new](https://crates.io/crates/derive-new) is from `@nrc` but doesn't seem to support `Into<X>` parameters, which is important for marker. It also doesn't seem like the crate isn't actively maintained, when I look at the [issue tracker](https://github.com/nrc/derive-new/issues) :thinking:

Reply:
I'd recommend you to use either [`typed-builder`](https://github.com/idanarye/rust-typed-builder), for more elaborate cases: [`buildstructor`](https://docs.rs/buildstructor/latest/buildstructor/)

Reply:
I'll try using `typed-builder` instead of the newly added constructors, wish me luck!

Reply:
It seems to work quite well. I won't update everything rn, but I'll keep the dependency and use it for new types :)

Here are the changes: https://github.com/rust-marker/marker/pull/244/commits/367cfb5680ed43f597b8b6edce64a08d06ad6c31

Reply:
There was also another lint that caused problems `clippy::used_underscore_binding` (https://github.com/idanarye/rust-typed-builder/issues/113)

Let's hope everything works now :)